### PR TITLE
Introducing TimeSeries Aspects + Dataset Profile (Stats) Aspect

### PR DESCRIPTION
**Pull request by jjoyce0510**
**Summary**
This PR include a host of changes from the Acryl team, including what was demo'd at the July Town Hall. 

**Changes**
This PR includes a host of changes, including

- Introduction of a new concept: TimeSeries Aspects. See documentation on Metadata Modeling for more information. 
- Introduction of a new TimeSeries Aspect: Dataset Profile. This represents a timestamped bundle of statistics about a particular dataset. 
- Introduction of MetadataChangeProposal and MetadataChangeLog events, the more generic & appropriately named future of MAE + MCE. 
- Support for extracting Dataset Profiles from SQL sources in Metadata Ingestion Framework. 
- Miscellaneous Documentation Additions + Improvements. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/CostValueMapper.java [hunk 1]
```diff
@@ -16,7 +16,7 @@ public static CostValue map(@NonNull final com.linkedin.common.CostValue costVal
     public CostValue apply(@NonNull final com.linkedin.common.CostValue costValue) {
         final CostValue result = new CostValue();
         result.setCostCode(costValue.getCostCode());
-        result.setCostId(costValue.getCostId());
+        result.setCostId(costValue.getCostId().floatValue());
```
Comment (dexter-mh-lee) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/CostValueMapper.java:19]: Is this an intended change? @jjoyce0510

File: datahub-graphql-core/src/main/resources/gms.graphql [hunk 2]
```diff
@@ -2908,3 +2926,36 @@ enum TimeRange {
...
+  durationMillis: Long!
+}
+
+type PartitionSpec {
+  partition: String!
+  timePartition: TimeWindow
+}
+
+type DatasetProfile implements TimeSeriesAspect {
+  timestampMillis: Long!
```
Comment (dexter-mh-lee) [inline datahub-graphql-core/src/main/resources/gms.graphql:2955]: so even if you implement an interface, do you still need to declare this?

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/CostValueMapper.java (hunk 1 above)
Comment (jjoyce0510) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/CostValueMapper.java:19]: Yes - These are represented as floats in the GQL model

File: datahub-graphql-core/src/main/resources/gms.graphql (hunk 2 above)
Comment (jjoyce0510) [reply datahub-graphql-core/src/main/resources/gms.graphql:2955]: Unfortunately, yes

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!!

**Review by shirshanka — APPROVED** [review]

**Pull request merged**